planner/go_planner.go

```diff
@@ -22,14 +22,20 @@ func (g *GoPlanner) IsRelevant(srcDir string) bool {
 	return fileExists(goModPath)
 }
 
-func (g *GoPlanner) Plan(srcDir string) *BuildPlan {
-	return &BuildPlan{
+func (g *GoPlanner) GetPlan(srcDir string) *Plan {
+	return &Plan{
 		Packages: []string{
 			"go",
 		},
-		InstallCommand: "go get",
-		BuildCommand:   "CGO_ENABLED=0 go build -o out",
-		StartCommand:   "./out", // TODO: Move gin specific stuff elsewhere.
+		InstallStep: &Step{
```

Comment:
For packages, will mergo append `go` on top of the `packages` definition in `devbox.json` config?

Reply:
If so, should we delete the `go` package in `go_planner`?

Reply:
Not sure I followed everything.

+ Packages will get appended and duplicates removed (see planner/plan_test.go)
+ We need `go` in order to call `go get` even in the case the user has not requested that package.

Is that what you were asking?

Reply:
That would cause an error (see https://github.com/jetpack-io/devbox/issues/33), but I believe the fix is to improve the planner code to detect the version of go (which is stated in the `go.mod`), and add the right package

Reply:
That would not work for NodeJS. I'm guessing for Node I can just use the user specified one.